### ISSUE TITLE
[Bug]: JSONPath wilcard selector not supported?

### ISSUE THREAD

**Issue by RemiBardon** [stepci/stepci#226]
### What happened?

I make an API call which returns

```json
[
  { "jid": "Margarett25@example.org", "role": "ADMIN" },
  { "jid": "Stanford.Simonis93@example.org", "role": "MEMBER" },
  { "jid": "Samara_Prohaska59@example.org", "role": "MEMBER" }
]
```

I'd like to capture the `JID`s, so I added

```yaml
captures:
  allMembersJids:
    jsonpath: $[*].jid
```

Unfortunately, it only captured one element:

```json
{
  "allMembersJids": "Margarett25@example.org"
}
```

### What did you expect to happen?

The [wildcard selector](https://www.rfc-editor.org/rfc/rfc9535#name-wildcard-selector) should match all array elements and I should get:

```json
{
  "allMembersJids": [
    "Margarett25@example.org",
    "Stanford.Simonis93@example.org",
    "Samara_Prohaska59@example.org"
  ]
}
```

### Version

2.8.1

### Environment

Node v22.2.0

### How can we reproduce this bug?

```yaml
version: "1.1"
name: Bug report
tests:
  example:
    name: Example jsonpath wildcard usage
    steps:
      - name: List pets
        http:
          method: GET
          url: https://petstore.swagger.io/v2/pet/findByStatus
          params:
            status: pending
          check:
            status: 200
          captures:
            values:
              jsonpath: $[*].name
            count:
              jsonpath: $.length
      - name: Test capture
        http:
          method: POST
          url: https://echo.free.beeceptor.com
          # `captures.values` isn't even an array, so it won't execute correctly
          body: ${{ captures.values }}
          headers:
            Content-Type: application/json
          check:
            jsonpath:
              $.parsedBody.ids.length: ${{ captures.count }}
```

### Would you be interested in working on a bugfix for this issue?

- [ ] Yes! Assign me

**Comment by GaneshSelvarasu** [comment]
I ran into the same problem. I noticed that only the first element from the jsonpath result is sent to checkResult() which is still OK when jsonpath expression is evaluated to a single element. But when dealing with wildcards, the entire result array shall be passed to checkResult(). As a workaround, I have something like below:

```
const result = JSONPath({ path, json })
stepResult.checks.jsonpath[path] = checkResult(
    (result.length === 1) ? result[0] : result,
    params.check.jsonpath[path]
);
```

Happy to contribute if this is OK.					)
